internal/assets/generated_flannel_manifest.go

```diff
@@ -143,22 +144,36 @@ metadata:
     tier: node
     app: flannel
 spec:
+  selector:
+    matchLabels:
+      app: flannel
   template:
     metadata:
       labels:
         tier: node
         app: flannel
     spec:
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+              - matchExpressions:
+                  - key: beta.kubernetes.io/os
+                    operator: In
+                    values:
+                      - linux
+                  - key: beta.kubernetes.io/arch
+                    operator: In
+                    values:
+                      - amd64
       hostNetwork: true
-      nodeSelector:
-        beta.kubernetes.io/arch: amd64
       tolerations:
         - operator: Exists
           effect: NoSchedule
       serviceAccountName: flannel
       initContainers:
         - name: install-cni
-          image: quay.io/coreos/flannel:{{.flannel_image_version}}-amd64
```

Comment:
Maybe you want to keep this templating?

Reply:
I've removed the hardcoded values here: https://github.com/DGollings/terraform-provider-kubeadm/pull/1

Reply:
Yes, the templates must be kept in order to configure things...

Reply:
I can make a new PR with @DGollings changes and mine if you want

Reply:
https://github.com/inercia/terraform-provider-kubeadm/pull/40

Reply:
@pablokbs apologies for the dealy in my response. That would be perfect, I think that the only requirement would be to keep these files as templates, as the provider uses them for replacing all the values configured in Terraform...